packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -112,9 +112,16 @@ export class DefaultCrudRepository<T extends Entity, ID>
     // We need to convert property definitions from PropertyDefinition
     // to plain data object because of a juggler limitation
     const properties: {[name: string]: object} = {};
-    for (const p in definition.properties) {
-      properties[p] = Object.assign({}, definition.properties[p]);
-    }
+
+    // We need to convert PropertyDefinition into the definition that
+    // the juggler understands
+    Object.entries(definition.properties).forEach(([key, value]) => {
+      if (value.type === 'array' || value.type === Array) {
+        value = Object.assign({}, value, {type: [value.itemType]});
+        delete value.itemType;
```

Comment:
Do we need to delete `value.itemType`? This will get deleted from the actual `definition.properties` Object IIRC -- in case it needs to be referenced again.

Reply:
Would it? I did want to avoid the manipulation of actual `definition.properties` by reassigning the variable `value` to what is a modified clone of `value` in line 120. Did I miss a step? This test case should cover it: https://github.com/strongloop/loopback-next/pull/1570/files#diff-328c8e879ea3a53d7fb50599f1c6a77eR121

Reply:
nvm, the `Object.assign` creates a new reference. This is fine.